geth/jail/internal/timers/timers.go

```diff
@@ -9,62 +9,81 @@ import (
 	"github.com/status-im/status-go/geth/jail/internal/vm"
 )
 
-var minDelay = map[bool]int64{
-	true:  10,
-	false: 4,
-}
+type intervalFunc func(*loop.Loop) func(call otto.FunctionCall) otto.Value
 
-//Define jail timers
-func Define(vm *vm.VM, l *loop.Loop) error {
+// Define jail timers
+func Define(vm *vm.VM, l *loop.Loop) (err error) {
 	if v, err := vm.Get("setTimeout"); err != nil {
 		return err
 	} else if !v.IsUndefined() {
 		return nil
 	}
 
-	newTimer := func(interval bool) func(call otto.FunctionCall) otto.Value {
-		return func(call otto.FunctionCall) otto.Value {
-			delay, _ := call.Argument(1).ToInteger()
-			if delay < minDelay[interval] {
-				delay = minDelay[interval]
-			}
-
-			t := &timerTask{
-				duration: time.Duration(delay) * time.Millisecond,
-				call:     call,
-				interval: interval,
-			}
-
-			// If err is non-nil, then the loop is closed and should not
-			// be used anymore.
-			if err := l.Add(t); err != nil {
-				return otto.UndefinedValue()
-			}
-
-			t.timer = time.AfterFunc(t.duration, func() {
-				l.Ready(t) // nolint: errcheck
-			})
-
-			value, newTimerErr := call.Otto.ToValue(t)
-			if newTimerErr != nil {
-				panic(newTimerErr)
-			}
-
-			return value
+	if err = vm.Set("setInterval", newTimer(true, l)); err != nil {
+		return
+	}
+	if err = vm.Set("setTimeout", newTimer(false, l)); err != nil {
+		return
+	}
+
+	timers := map[string]intervalFunc{
```

Comment:
It was the initial idea, however `newTimer` function has an extra bool parameter (interval)  and because of this it doesn't accomplish the `intervalFunc` interface.

My initial approach was to always send the interval parameter even if `immediatetimer` and `clearTimeout` don't use it, so all functions accomplish the same interface. If we do this its possible to group them all on the same map. 

Even resulting code looks clean, i wasn't sure it was easy to understand for someone not familiar with the code. If you're happy with that approach I can easily implement it, please let me know what to do :)

Reply:
Ok, now I have the complete picture :) Up to you, the current approach, in my opinion, is fine as well. 

If you'd like to group all functions in the map and just do `func someFunc(_ bool, l *loop.Loop)` then it's also ok. I just recommend changing the order of parameters as `l *loop.Loop` is required in each function.